kotlin/generate-offline-map/src/main/java/com/esri/arcgisruntime/sample/generateofflinemap/MainActivity.kt

```diff
@@ -0,0 +1,231 @@
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.concurrent.Job
+import com.esri.arcgisruntime.geometry.Envelope
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.security.AuthenticationManager
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapJob
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapParameters
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapTask
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.dialog_layout.*
+import java.io.File
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG = MainActivity::class.java.simpleName
+
+  private val graphicsOverlay: GraphicsOverlay by lazy {
+    GraphicsOverlay().also {
+      // add the graphics overlay to the map view when it is created
+      mapView.graphicsOverlays.add(it)
+    }
+  }
+
+  private val downloadArea: Graphic by lazy {
+    Graphic().also {
+      // create a symbol to show a box around the extent we want to download
+      it.symbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.RED, 2F)
+      // add the graphic to the graphics overlay when it is created
+      graphicsOverlay.graphics.add(it)
+    }
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // handle authentication with the portal
+    AuthenticationManager.setAuthenticationChallengeHandler(
+      DefaultAuthenticationChallengeHandler(this)
+    )
+
+    // disable the button until the map is loaded
+    takeMapOfflineButton.isEnabled = false
+
+    // create a portal item with the itemId of the web map
+    val portal = Portal(getString(R.string.portal_url), false)
+    val portalItem = PortalItem(portal, getString(R.string.item_id))
+    // create a map with the portal item
+    val map = ArcGISMap(portalItem)
+
+    map.addDoneLoadingListener {
+      if (map.loadStatus == LoadStatus.LOADED) {
+        // limit the map scale to the largest layer scale
+        map.apply {
+          maxScale = operationalLayers[6].maxScale
+          minScale = operationalLayers[6].minScale
+        }
+      } else {
+        val error = "Map failed to load: " + map.loadError.message
+        Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+        Log.e(TAG, error)
+      }
+    }
+
+    // set the map to the map view
+    mapView.map = map
+
+    // update the download area box whenever the viewpoint changes
+    mapView.addViewpointChangedListener {
```

Comment:
I've encountered this twice now, so it might be worth discussing with you and @Rachael-E :

`mapView.map = map` looks nice and clean, but if mapView stuff is moved into an apply block, then it becomes `map = map` which is a compiler error. It's easy to solve if the map creation is just one line, but if the map also needs setup (ie. its own apply block), the solutions are to either:

1. Rename the map val to something else, ie. `map = webMap` or whatever is most appropriate.
2. use `this.map`, which is accurate, but I feel using `this` can get a bit confusing and might better be avoided.
3. nest the `map.apply{}` inside the `mapView.apply{}` ie.
```kotlin
mapView.apply {
  map = ArcGISMap(portalItem).apply{
    addDoneLoadingListener { .. }
    maxScale = ...
  }
  addViewpointChangedListener { ... }
}
```
What do y'all think?

Reply:
For now I'm using `this.map = map`!

Reply:
I like this.map = map

And we're trying to not nest applies as the reference to `this` can become unclear

Reply:
So I'd say 1 or 2 👍